### PR TITLE
[Proposal] Standardize errors and Logs with Middlewares

### PR DESCRIPTION
**Pull request by drgarcia1986**
Only applied for `Deploy` pkg.

---
This change is [Reviewable](https://reviewable.io/reviews/luizalabs/teresa-api/214)

### CODE REVIEW

File: pkg/server/k8s/client.go [hunk 1]
```diff
@@ -366,15 +366,17 @@ func (k *k8sClient) HasService(namespace, appName string) (bool, error) {
...
 
 func (k *k8sClient) CreateService(namespace, appName string) error {
 	srvSpec := serviceSpec(namespace, appName, k.defaultServiceType)
-	_, err := k.kc.CoreV1().Services(namespace).Create(srvSpec)
-	return err
+	if _, err := k.kc.CoreV1().Services(namespace).Create(srvSpec); err != nil {
+		return errors.Wrap(err, "create service failed")
+	}
+	return nil
 }
```
Comment (aguerra) [inline pkg/server/k8s/client.go]: I guess you can write:
```go
_, err := k.kc.CoreV1().Services(namespace).Create(srvSpec)
return errors.Wrap(err, "create service failed")
```

File: pkg/server/deploy/handlers.go [hunk 2]
```diff
@@ -54,7 +55,7 @@ func (s *Service) Make(stream dpb.Deploy_MakeServer) error {
 	rs := bytes.NewReader(content.Bytes())
 	rc, err := s.ops.Deploy(u, appName, rs, description, s.options.RevisionHistoryLimit)
 	if err != nil {
-		return err
+		return teresa_errors.Get(err)
```
Comment (aguerra) [inline pkg/server/deploy/handlers.go]: I thought you could `return err` here because the log middleware does the unwrapping, am I right?

Comment (drgarcia1986) [inline pkg/server/deploy/handlers.go]: you're right!

File: pkg/server/k8s/client.go (hunk 1 above)
Comment (drgarcia1986) [inline pkg/server/k8s/client.go]: `errors.Wraps` returns `nil` for non errors (i.e. `nil`) args ?

Comment (aguerra) [inline pkg/server/k8s/client.go]: yep

Comment (drgarcia1986) [inline pkg/server/k8s/client.go]: awesome

File: pkg/server/middlewares.go [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+}
+
+func logStreamInterceptor(srv interface{}, stream grpc.ServerStream, info *grpc.StreamServerInfo, handler grpc.StreamHandler) error {
+	err := handler(srv, stream)
+	if err != nil {
+		log.WithField("route", info.FullMethod).Error(err)
+		return teresa_errors.Get(err)
+	}
+	return nil
+}
```
Comment (aguerra) [inline pkg/server/middlewares.go]: Maybe Error -> WithError

Comment (drgarcia1986) [inline pkg/server/middlewares.go]: I changed

Comment (drgarcia1986) [inline pkg/server/middlewares.go]: Also, I added the user of the request. Check some examples below and let know if you liked.

`time="2017-07-18T14:32:31Z" level=error msg="Log Interceptor got an Error" error="namespaces \"hashpass2\" already exists" request=name:"hashpass2" team:"arquitetura" limits:<default:<quantity:"500m" resource:"cpu" > default:<quantity:"512Mi" resource:"memory" > default_request:<quantity:"200m" resource:"cpu" > default_request:<quantity:"512Mi" resource:"memory" > > auto_scale:<cpu_target_utilization:70 max:2 min:1 >  route="/app.App/Create" user="diego.garcia@luizalabs.com"`

`time="2017-07-18T14:33:17Z" level=error msg="Log Interceptor got an Error" error="rpc error: code = NotFound desc = App not found: get annotation failed: namespaces \"hashpass3\" not found" route="/deploy.Deploy/Make" user="diego.garcia@luizalabs.com"`

ps: In the unary interceptor, I'm logging request content.

**Pull request merged**